forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +273,153 @@ impl WalletCommands {
         }
     }
 }
+
+fn format_balance_string(num: Decimal, mode: FormattingMode) -> anyhow::Result<String> {
```

Comment:
How about we change this function to take a `TokenAmount` and then make it fail if we cannot convert to a `Decimal`? This failure should be tested.

Reply:
I already propagate the parsing error on line 211 in code. And test the parsing on line 418 in tests.

Reply:
You would get better type safety if you used `TokenAmount`. The code would even be slightly more self-documenting.

Reply:
Done this